filesystems/watcher.go

```diff
@@ -578,27 +617,19 @@ func (w *Watcher) uploadAllLogFiles(ctx context.Context) error {
 		return err
 	}
 
-	var errs *multierror.Error
-
-	for _, logFile := range logFileList {
-		if ctx.Err() != nil {
-			errs = multierror.Append(errs, ctx.Err())
-			break
-		}
-		if CheckFileForStdio(logFile) {
-			continue
+	//Remove duplicate entries in the list of files to upload
+	filenames := make(map[string]bool)
+	uniqueFiles := []string{}
```

Comment:
Do we actually get dupes here? Also, any reason not to use sets:
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets

Reply:
Since this is the entire list of files we upload, IMO if we catch it here we should not have any duplicates. Catching duplicates now with sets.